lib/waterline/core/typecast.js

```diff
@@ -87,11 +88,14 @@ Cast.prototype.run = function(values) {
         break;
 
       case 'date':
-      case 'time':
```

Comment:
Should `time` have a different logic than `datetime` and `date`? That looks like an API change.